corehq/apps/userreports/sql/adapter.py

```diff
@@ -260,7 +261,57 @@ def doc_exists(self, doc):
             return session.query(query.exists()).scalar()
 
 
-class ErrorRaisingIndicatorSqlAdapter(IndicatorSqlAdapter):
+class MultiDBSqlAdapter(IndicatorSqlAdapter):
```

Comment:
This is strange class structure since it's a mix of inheritance and composition. I think it would be better to stick to composition (like [MigratingBlobDB](https://github.com/dimagi/commcare-hq/blob/725265bcc8829a9e010f0115033b3c5e1eeeb6f4/corehq/blobs/migratingdb.py#L9)).

That would also simplify the logic in `get_indicator_adapter` and also mean that you wouldn't need to have `ErrorRaisingMultiDBAdapter` as well.

Reply:
Thanks, I will try to address this

Reply:
Updated, the original way was more DRY, though agree, it could be confusing.